dizoo/taxi/envs/taxi_env.py

```diff
@@ -0,0 +1,166 @@
+        np.random.seed(self._seed)
+        
+    def step(self, action: np.ndarray) -> BaseEnvTimestep:
+        assert isinstance(action, np.ndarray), type(action)
+        action = action.item()
+        obs, rew, done, info = self._env.step(action)
+        self._eval_episode_return += rew
+        obs = self._encode_taxi(obs)
+        rew = to_ndarray([rew])  # Transformed to an array with shape (1, )
+        if self._save_replay:
+            picture = self._env.render()
+            self._frames.append(picture)
+        if done:
+            info['eval_episode_return'] = self._eval_episode_return
+            if self._save_replay:
+                assert self._replay_path is not None, "your should have a path"
+                path = os.path.join(
+                    self._replay_path, '{}_episode_{}.gif'.format(self._cfg.env_id, self._save_replay_count)
+                )
+                self.frames_to_gif(self._frames, path)
+                self._frames = []
+                self._save_replay_count += 1
+        rew = rew.astype(np.float32)
+        obs = obs.astype(np.float32)
+        return BaseEnvTimestep(obs, rew, done, info)
+    
+    def enable_save_replay(self, replay_path: Optional[str] = None) -> None:
+        if replay_path is None:
+            replay_path = './video'
+            if not os.path.exists(replay_path):
+                os.makedirs(replay_path)
+        self._replay_path = replay_path
+        self._save_replay = True
+        self._save_replay_count = 0
+        
+    def random_action(self) -> np.ndarray:
+        random_action = self.action_space.sample()
+        if isinstance(random_action, np.ndarray):
+            pass
+        elif isinstance(random_action, int):
+            random_action = to_ndarray([random_action], dtype=np.int64)
+        elif isinstance(random_action, dict):
+            random_action = to_ndarray(random_action)
+        else:
+            raise TypeError(
+                '`random_action` should be either int/np.ndarray or dict of int/np.ndarray, but get {}: {}'.format(
+                type(random_action), random_action
+                )
+            )
+        return random_action
+        
+    #todo encode the state into a vector    
+    def _encode_taxi(self, obs: np.ndarray) -> np.ndarray:
+        taxi_row, taxi_col, passenger_location, destination = self._env.unwrapped.decode(obs)
+        return to_ndarray([taxi_row, taxi_col, passenger_location, destination])
+        
+    @property
+    def observation_space(self) -> Space:
+        return self._observation_space
+
+    @property
+    def action_space(self) -> Space:
+        return self._action_space
+
+    @property
+    def reward_space(self) -> Space:
+        return self._reward_space
+    
+    def __repr__(self) -> str:
+        return "DI-engine Taxi-v3 Env"
+    
+    @staticmethod
+    def frames_to_gif(frames: List[imageio.core.util.Array], gif_path: str, duration: float = 0.1) -> None:
+        """

```

Comment:
Polish argument list as the format in https://github.com/opendilab/DI-engine/blob/main/ding/model/common/encoder.py.